src/build/cargo.rs

```diff
@@ -299,20 +308,29 @@ impl Executor for RlsExecutor {
             }
         }
 
-        if self.workspace_mode {
-            let output = cmd.output().expect("Couldn't execute rustc");
-            let mut stderr_json_msg = convert_message_to_json_strings(output.stderr);
-            self.compiler_messages.lock().unwrap().append(&mut stderr_json_msg);
-        } else {
-            cmd.status().expect("Couldn't execute rustc");
-        }
-
         // Finally, store the modified cargo-generated args/envs for future rustc calls
         args.insert(0, rustc_exe);
```

Comment:
we don't need to pass it, but it is so that the args look like what would happen if you took the args from inside the program (there you get the executable name as the first arg). We could not insert it and strip that arg if we ever still take the args from inside the program (I'm not sure we do).